uwrt_mars_rover_drivetrain/uwrt_mars_rover_drivetrain_hw/src/uwrt_mars_rover_drivetrain_hw_actuator_interface.cpp

```diff
@@ -0,0 +1,188 @@
+#include "uwrt_mars_rover_drivetrain_hw/uwrt_mars_rover_drivetrain_hw_actuator_interface.hpp"
+
+namespace uwrt_mars_rover_drivetrain_hw {
+LifecyleNodeCallbackReturn UWRTMarsRoverDrivetrainHWActuatorInterface::on_init(
+    const hardware_interface::HardwareInfo& actuator_info) {
+  RCLCPP_DEBUG(rclcpp::get_logger("UWRTMarsRoverDrivetrainHWActuatorInterface"), "Drivetrain Actuator Initializing...");
+
+  if (hardware_interface::ActuatorInterface::on_init(actuator_info) != LifecyleNodeCallbackReturn::SUCCESS) {
+    return LifecyleNodeCallbackReturn::ERROR;
+  }
+
+  // Validate that parsed urdf information matches expected structure
+  if (info_.joints.size() != NUM_JOINTS) {
+    RCLCPP_FATAL_STREAM(
+        rclcpp::get_logger("UWRTMarsRoverDrivetrainHWActuatorInterface"),
+        "'" << info_.name.c_str() << "' has " << info_.joints.size() << " joints. " << NUM_JOINTS << " expected.");
+    return LifecyleNodeCallbackReturn::ERROR;
+  }
+  const hardware_interface::ComponentInfo& joint = info_.joints[0];
+  if (joint.state_interfaces.size() != NUM_STATE_INTERFACES) {
+    RCLCPP_FATAL_STREAM(rclcpp::get_logger("UWRTMarsRoverDrivetrainHWActuatorInterface"),
+                        "Joint '" << joint.name.c_str() << "' has " << joint.state_interfaces.size()
+                                  << " state interfaces. " << NUM_STATE_INTERFACES << " expected.");
+    return LifecyleNodeCallbackReturn::ERROR;
+  }
+  for (const hardware_interface::InterfaceInfo& state_interface : joint.state_interfaces) {
+    if (!(state_interface.name == hardware_interface::HW_IF_POSITION ||
+          state_interface.name == hardware_interface::HW_IF_VELOCITY ||
+          state_interface.name == "iq_current")) {  // TODO: add shared HW_IF_IQ_CURRENT in some shared package
```

Comment:
lol I was thinking same .... I just defined this in my class
`static constexpr char HW_IF_IQ_CURRENT[] = "iq_current;"` .... just copy and paste from what ros2 does

Reply:
yeah. I think its best to keep it as just a string until we create a package for it to avoid duplicate definitions everywhere. Probably like a top level `uwrt_mars_rover_hw` package like what i did for shared custom interfaces in the `ros1` branch

Reply:
#179